### PR TITLE
feat: Native UI `Spinner`

### PR DESCRIPTION
**Pull request by jordmccord**
Adds the `Spinner` component to the library. Also introduces the use of the `react-native-reanimated` library.

- [Spinner Figma Designs](https://www.figma.com/file/3RY3OvLA88yZksRjOfjQJm/UW-App-UI?type=design&node-id=5627-1530&mode=design)
- [React Native Reanimated](https://docs.swmansion.com/react-native-reanimated/)

### CODE REVIEW

File: packages/native-ui/src/components/custom/Spinner/Spinner.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    default:
+      return 32;
+  }
+};
+
+const Spinner: React.FC<SpinnerProps> = ({ size = 'md', color }) => {
+  const width = getWidth(size);
+  const CIRCUMFERENCE = (width - 4) * Math.PI;
+  const R = CIRCUMFERENCE / (2 * Math.PI);
+  const STROKE_WIDTH = 2;
```
Comment (noeliacabane) [inline packages/native-ui/src/components/custom/Spinner/Spinner.tsx:39]: I think this is the one that the size makes it conditional. 

> so it's 2px on > 16px and 1.5px on < 16px

(from slack) this is probably what @rlanguw saw

File: packages/native-ui/src/components/custom/Spinner/Spinner.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        viewBox={`0 0 ${DIAMETER} ${DIAMETER}`}
+        style={Platform.OS === 'web' ? {} : animatedSvgStyle}
+        animatedProps={Platform.OS === 'web' ? animatedSvgProps : {}}
+      >
+        <G origin={`${HALF_CIRCLE}, ${HALF_CIRCLE}`} rotation={-90}>
+          <StyledCircle
+            fill="transparent"
+            stroke={color ? color : 'currentColor'}
+            strokeWidth={STROKE_WIDTH}
+            cx="50%"
```
Comment (noeliacabane) [inline packages/native-ui/src/components/custom/Spinner/Spinner.tsx:103]: Interesting to see that there is not a shorthand for this prop 🤔 😓 

maybe is something we could put in a global variable in the future for where we want a circles within an SVG component?

Comment (jordmccord) [reply packages/native-ui/src/components/custom/Spinner/Spinner.tsx:103]: Which prop?

Comment (noeliacabane) [reply packages/native-ui/src/components/custom/Spinner/Spinner.tsx:103]: The snippet that generates the circle

```
cx="50%"
cy="50%"
r={R}
```

**Review by noeliacabane — APPROVED** [review]
Looks good, left some comments that are not blocking

**Review by UWlouisehenry — APPROVED** [review]

**Pull request merged**